File: Algorithms/Sorting/MarkAndToys/MarkAndToys.go
package main

import (
	"fmt"
	"sort"
)

func main() {
	var n, k, count, i, amount int
	fmt.Scan(&n)
	fmt.Scan(&k)
	toys := make([]int, n)
	for i := 0; i < n; i++ {
		fmt.Scan(&toys[i])
	}
	sort.Ints(toys)
	for amount+toys[i] < k && i < n {
		amount += toys[i]
		count++
		i++
	}
	fmt.Println(count)
}

File: Algorithms/Sorting/MarkAndToys/MarkAndToys.cs
using System;

class MarkAndToys {
  static void Main(string[] args) {
    int[] nk = Array.ConvertAll(Console.ReadLine().TrimEnd().Split(' '), e => Convert.ToInt32(e));
    int n = nk[0];
    int k = nk[1];
    int[] toys = Array.ConvertAll(Console.ReadLine().TrimEnd().Split(' '), e => Convert.ToInt32(e));
    int amount = 0, i = 0, count = 0;
    Array.Sort(toys);
    while (amount + toys[i] < k && i < n) {
      amount += toys[i];
      count++;
      i++;
    }
    Console.WriteLine(count);
  }
}